src/signer.rs

```diff
@@ -136,9 +149,77 @@ impl CngSigner {
             ),
             _ => return Err(Error::General("Unsupported signature scheme".to_owned())),
         };
+
         Ok((hash, padding))
     }
+*/
+    // new hash function using BCryptHash
+    fn hash(&self, message: &[u8]) -> Result<(Vec<u8>, SignaturePadding), Error> {
+        let (alg, padding) = match self.scheme {
+            SignatureScheme::RSA_PKCS1_SHA256 => (
+                BCRYPT_SHA256_ALG_HANDLE,
+                SignaturePadding::Pkcs1,
+            ),
+            SignatureScheme::RSA_PKCS1_SHA384 => (
+                BCRYPT_SHA384_ALG_HANDLE,
+                SignaturePadding::Pkcs1,
+            ),
+            SignatureScheme::RSA_PKCS1_SHA512 => (
+                BCRYPT_SHA512_ALG_HANDLE,
+                SignaturePadding::Pkcs1,
+            ),
+            SignatureScheme::RSA_PSS_SHA256 => (
+                BCRYPT_SHA256_ALG_HANDLE,
+                SignaturePadding::Pss,
+            ),
+            SignatureScheme::RSA_PSS_SHA384 => (
+                BCRYPT_SHA384_ALG_HANDLE,
+                SignaturePadding::Pss,
+            ),
+            SignatureScheme::RSA_PSS_SHA512 => (
+                BCRYPT_SHA512_ALG_HANDLE,
+                SignaturePadding::Pss,
+            ),
+            SignatureScheme::ECDSA_NISTP256_SHA256 => (
+                BCRYPT_SHA256_ALG_HANDLE,
+                SignaturePadding::None,
+            ),
+            SignatureScheme::ECDSA_NISTP384_SHA384 => (
+                BCRYPT_SHA384_ALG_HANDLE,
+                SignaturePadding::None,
+            ),
+            _ => return Err(Error::General("Unsupported signature scheme".to_owned())),
+        };
+
+               
+        let hash_len = match alg { 
+            BCRYPT_SHA256_ALG_HANDLE => 32, 
+            BCRYPT_SHA384_ALG_HANDLE => 48, 
+            BCRYPT_SHA512_ALG_HANDLE => 64, 
```

Comment:
I would expect these hash lengths to already be defined in  windows_sys

Reply:
didn't find those length constant in window_sys.